tests/sentry/integrations/opsgenie/test_integration.py

```diff
@@ -82,3 +90,39 @@ def test_invalid_url(self):
         with pytest.raises(IntegrationError) as error:
             provider.get_account_info(base_url=bad_url, api_key=self.config["api_key"])
         assert str(error.value) == "Invalid URL provided."
+
+    @responses.activate
+    def test_update_config_valid(self):
+        integration = Integration.objects.create(
+            provider="opsgenie", name="test-app", external_id=EXTERNAL_ID, metadata=METADATA
+        )
+
+        integration.add_organization(self.organization, self.user)
+        installation = integration.get_installation(self.organization.id)
+        opsgenie_client = installation.get_client()
+
+        data = {"team_table": [{"id": "", "team": "cool-team", "integration_key": "1234-5678"}]}
+        resp_data = {"data": {"id": "123-id", "name": "cool-team"}}
+        responses.add(
+            responses.GET, url=f"{opsgenie_client.base_url}/teams/cool-team", json=resp_data
+        )
+
+        installation.update_organization_config(data)
+        assert installation.get_config_data() == {
+            "team_table": [{"id": "123-id", "team": "cool-team", "integration_key": "1234-5678"}]
+        }
+
+    def test_update_config_invalid(self):
+        integration = Integration.objects.create(
+            provider="opsgenie", name="test-app", external_id=EXTERNAL_ID, metadata=METADATA
+        )
+
+        integration.add_organization(self.organization, self.user)
+        installation = integration.get_installation(self.organization.id)
+        opsgenie_client = installation.get_client()
+
+        data = {"team_table": [{"id": "", "team": "cool-team", "integration_key": "1234-bad"}]}
+        responses.add(responses.GET, url=f"{opsgenie_client.base_url}/teams/cool-team")
+        with pytest.raises(ValidationError):
+            installation.update_organization_config(data)
+        assert installation.get_config_data() == {}
```

Comment:
This is optional but you can also assert the err msg if you want, helps sometimes to make sure you're catching the right thing and not some other random err
```suggestion
        with pytest.raises(ValidationError) as excinfo:
            installation.update_organization_config(data)
        assert exc.value == "some error message"
        assert installation.get_config_data() == {}
```